### PR TITLE
One on One Peer Review

### PR DESCRIPTION
**Pull request by sntakirutimana72**
### Instructions

- Once you get a pull request link from your partner, go ahead and review it:
    - Reminder: [How to give a code review](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/code-review/articles/give_code_review_basics.md).
- Check the quality of the code (you can use previous lessons as a guide). Remember to check for the following:
    - [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md ).
    - [DRY, KISS, and YAGNI](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/lesson_DRY_KISS_YAGNI.md) rules.
    - [HTML & CSS best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- Finish your review with a nice summary.
- Compare your review with the issue created by the project's author in the [Do I follow JavaScript best practices?](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/exercises/clean_code_self_assessment.md) exercise.
    - _There is no need to add anything to your review at this point. It is just to check if you both noticed the same issues._
- Check the review you received from your partner! You will make improvements to your code based on your partner's review in the next exercise.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                            <i class="fa-solid fa-rotate"></i>
+                        </button>
+                    </div>
+                    <div class="d-flex-align-center field-wrap row">
+                        <input type="text" name="desc" placeholder="Add to your list.." required>
+                        <span><i class="fa fa-level-down"></i></span>
+                    </div>
+                </form>
+                <ul class="d-flex-column to-do-list no-spacing"></ul>
+                <button type="button" class="to-do-clear-btn" disabled>Clear all completed</button>
```
Comment (cyrilmuchemi) [inline src/index.html]: Your class naming convention looks a bit long. Kindly try to use short and precise naming conventions

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+/**
+*
+* @param {String} selector
+* @param {HTMLElement} parentTree
+* @returns
+*/
+function $selectAll(selector, tree = document.body) {
+  return tree.querySelectorAll(selector);
+}
+
```
Comment (cyrilmuchemi) [inline src/index.js]: Good use of JQuery. Kindly try to use DOM in the global environment

**Review by cyrilmuchemi — CHANGES_REQUESTED** [review]
Hey @sntakirutimana72  :green_circle: 
- Good job so far. Keep up the good work
- Kindly fix the Linter errors
- Implement required changes

File: src/index.js (hunk 2 above)
Comment (sntakirutimana72) [reply src/index.js]: Hi @cyrilmuchemi 

this is not a JQuery property but rather a custom implementation to avoid repeating the same code over and over.

Comment (cyrilmuchemi) [reply src/index.js]: Okay. i appreciate the clarification :smiley:

**Review by cyrilmuchemi — APPROVED** [review]
### Congratulations :balloon: :balloon: 
- There is nothing to say other than it's time to merge it :rabbit: 

Cheers and happy coding :dancing_women:

**Pull request merged**